crates/compiler/test_mono/generated/match_on_result_with_uninhabited_error_branch.txt

```diff
@@ -0,0 +1,7 @@
+procedure Test.0 ():
+    let Test.5 : Str = "abc";
+    let Test.1 : [C [], C Str] = TagId(1) Test.5;
+    let Test.3 : Str = UnionAtIndex (Id 1) (Index 0) Test.1;
```

Comment:
It would be nice if we could get rid of the `C []` constructor in the layout - @folkertdev attempted this before, but we ran into some issues there previously. Maybe things have changed, though.